google-cloud-spanner/src/test/java/com/google/cloud/spanner/DatabaseTest.java

```diff
@@ -178,6 +198,41 @@ public void testBuildWithDatabaseDialect() {
     assertEquals(Dialect.GOOGLE_STANDARD_SQL, database.getDialect());
   }
 
+  @Test
+  public void testBuildWithProtoDescriptors() throws IOException {
+    final Database database =
+        dbClient
+            .newDatabaseBuilder(DatabaseId.of("my-project", "my-instance", "my-database"))
+            .setProtoDescriptors("com/google/cloud/spanner/descriptors.pb")
+            .build();
+
+    assertEquals(protoDescriptors, database.getProtoDescriptors());
+  }
+
+  @Test
+  public void testBuildWithProtoDescriptorsFromInputStream() throws IOException {
+    InputStream in =
+        getClass().getClassLoader().getResourceAsStream("com/google/cloud/spanner/descriptors.pb");
+    final Database database =
+        dbClient
+            .newDatabaseBuilder(DatabaseId.of("my-project", "my-instance", "my-database"))
+            .setProtoDescriptors(in)
+            .build();
+
+    assertEquals(protoDescriptors, database.getProtoDescriptors());
+  }
+
+  @Test
+  public void testBuildWithProtoDescriptorsFromByteArray() throws IOException {
+    final Database database =
+        dbClient
+            .newDatabaseBuilder(DatabaseId.of("my-project", "my-instance", "my-database"))
+            .setProtoDescriptors(protoDescriptorsByteArray)
+            .build();
+
+    assertEquals(protoDescriptors, database.getProtoDescriptors());
+  }
+
```

Comment:
We should add tests for exception cases (invalid inputstream, invalid file path, etc) as well.

Reply:
Done. Can you check?